packages/server-web/src/client/store.js

```diff
@@ -46,15 +46,34 @@ export const store = new Vuex.Store({
     commands(state, commands) {
       state.commands = commands;
     },
-    processEvent(state, event) {
-      if (event.position > state.currentEventPositon) {
-        state.allEvents.push(event);
-        state.currentEventPositon = event.position;
-        state.todos = replay(state.allEvents).todos;
-      }
+    currentEventPosition(state, position) {
+      state.currentEventPosition = position;
+    },
+    todos(state, newTodos) {
+      state.todos = newTodos;
     },
     changePassword(state, val) {
       state.commands.changePassword(val);
+    },
+    addEventToAllEvents(state, event) {
+      state.allEvents = state.allEvents = [...state.allEvents, event];
+    }
+  },
+  actions: {
+    processEvent(context, event) {
+      new Promise((resolve, reject) => {
```

Comment:
Can this be done with `async processEvent(context, event) { ... }`, without having the Promise + try/catch boilerplate?

Reply:
not sure